pkg/file/file.go

```diff
@@ -12,18 +12,21 @@ import (
 )
 
 // ExtractInstallerMetadata extracts the software name and version from the
-// installer file. The format of the installer is determined based on the
-// extension of the filename.
-func ExtractInstallerMetadata(filename string, b []byte) (name, version string, err error) {
+// installer file and returns them along with the sha256 hash of the bytes. The
+// format of the installer is determined based on the extension of the
+// filename.
+func ExtractInstallerMetadata(filename string, r io.Reader) (name, version string, shaSum []byte, err error) {
```

Comment:
@gillespi314 The function now returns the SHA-256 sum too. Note that it now takes a reader instead of the bytes of the contents, for some file types it will be a bit more efficient as the whole bytes will not have to be loaded in memory all at once (not possible for all file types).